code-challenges.test.js

```diff
@@ -21,36 +21,144 @@
 // Example input: 10
 // Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
 
+describe("fibonacciNumbers", () => {
+  it("takes in a number (greater than 2) and returns an array that length containing the numbers of the Fibonacci sequence", () => {
+    // Example input: 6
+    // Expected output: [1, 1, 2, 3, 5, 8]
+    // Example input: 10
+    // Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
+    expect(fibonacciNumbers(6)).toEqual([1, 1, 2, 3, 5, 8])
+    expect(fibonacciNumbers(10)).toEqual([1, 1, 2, 3, 5, 8, 13, 21, 34, 55])
+  })
+})
+
+// Pseudocode
+// Create a function called fibonacciNumbers that takes in a number
+// Iteration with a for loop
+// Access each index and perform addition
+// Use push method to add elments to end of array
+// Expected output: [1, 1, 2, 3, 5, 8]
+// Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
 
 // b) Create the function that makes the test pass.
 
+const fibonacciNumbers = (number) => {
+  let array = [1, 1]
+  for(let i = 1; i < number - 1; i++) {
+    if (number > 2){
+      array.push(array[i] + array[i - 1])
+    }
+  }
+  return array
+}
+
+// PASS  ./code-challenges.test.js
+//  fibonacciNumbers
+//    ✓ takes in a number (greater than 2) and returns an array that length containing the numbers of the Fibonacci sequence (1 ms)
 
 // --------------------2) Create a function that takes in an array and returns a new array of only odd numbers sorted from least to greatest.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
+// const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
 // Expected output: [-9, 7, 9, 199]
 
-const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
+// const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
 // Expected output: [-823, 7, 23]
 
+describe("oddNumbersFiltered", () => {
+  it("takes in an array and returns a new array of only odd numbers sorted from least to greatest", () => {
+    const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
+    // Expected output: [-9, 7, 9, 199]
+    const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
+    // Expected output: [-823, 7, 23]
+    expect(oddNumbersFiltered(fullArr1)).toEqual([-9, 7, 9, 199])
+    expect(oddNumbersFiltered(fullArr2)).toEqual([-823, 7, 23])
+  })
+})
+
+// FAIL  ./code-challenges.test.js
+//  oddNumbersFiltered
+//    ✕ takes in an array and returns a new array of only odd numbers sorted from least to greatest
+//
+//  ● oddNumbersFiltered › takes in an array and returns a new array of only odd numbers sorted from least to greatest
+//
+//    ReferenceError: oddNumbersFiltered is not defined
+
+// Pseudocode
+// Create a function called oddNumbersFiltered that takes in an array.
+// Use filter to create a new array
+// Use if statement to iterate
```

Comment:
The filter is doing the iteration.  If statements are to check a condition which filter actually has built in to it.